[CS2113-T11-1] MediTask

AjayShanker-geek opened the pull request:
MediTask is a desktop application designed to help nurses efficiently manage and track their daily tasks. This CLI tool streamlines task management, allowing nurses to quickly organize, monitor, and complete tasks, ensuring no important steps are missed.

Chiang-HuiXin commented on docs/diagrams/ParserClass.puml:
<img width="893" alt="image" src="https://github.com/user-attachments/assets/ea831872-0cd2-4854-91d2-a5faa5b9ac35">
Non standard notation, should not have the "C", "I" at the method names according to prof.
The red box and green circle for visibility are non standard notations too. Otherwise, the diagram looks good.

Chiang-HuiXin commented on docs/diagrams/AddPatientSequenceDiagram.puml:
```diff
@@ -0,0 +1,35 @@
+    Model --> AddPatientCommand : Returns false (if no duplicate)
+    deactivate Model
+
+    AddPatientCommand -> Model : addPatient("John Doe")
```
I notice that there is no activation bar for addPatient() and saveHospitalData(). It is required to be consistent, if activation bar is used, then it should be used for all
<img width="40" alt="image" src="https://github.com/user-attachments/assets/17639ab4-79fe-44c1-933e-0201eb13a3dc">

Chiang-HuiXin commented on docs/diagrams/AddPatientSequenceDiagram.puml:
```diff
@@ -0,0 +1,35 @@
+    activate Storage
+    deactivate Storage
+
+    AddPatientCommand -> UI : Display success message
+else State is not MAIN_STATE
+    Parser -> UI : Display error message
```
The arrows should be dotted, otherwise the diagram looks detailed 👍

Chiang-HuiXin commented on docs/diagrams/AddPatientSequenceDiagram.puml:
<img width="676" alt="image" src="https://github.com/user-attachments/assets/c03c0fa6-1a89-4399-ba55-a3d2d8003bbb">

I realised that there are no colons at the entities, I think they are required?

Chiang-HuiXin reviewed (commented):
Overall, I think the diagrams are clear and detailed. 
<img width="216" alt="image" src="https://github.com/user-attachments/assets/c95f9fcd-306b-4a32-9944-52a47b7fa95d">
I also noticed that these 2 diagram cannot be viewed.

Dri-water commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,209 @@
+---
+
+## Setting up, getting started
+Refer to the guide for 'Setting up and getting started.'
```
Would be good if some details were still provided here

Dri-water commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,209 @@
+
+Given below is a quick overview of main components and how they interact with each other.
+
+**Main Components**
```
I like the overview of the important components. Great!

Dri-water commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,209 @@
+
+### Parser
+The Parser interface uses a series of classes to implement the various commands.
+![Parser_Class_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/ParserClass.png)
```
Very nice class diagram for visualisation!

Dri-water commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,209 @@
+
+![Storage_Class_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/StorageClassDiagram.png)
+
+**API** : [`StorageFile.java`](https://github.com/AY2425S1-CS2113-T11-1/tp/tree/master/src/main/java/seedu/duke/storage/StorageFile.java)
```
Very nice that the API is specified here, good job!

Dri-water commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,209 @@
 
+The following sequence diagram illustrates how the `AddPatientCommand` is executed:
+
+![AddPatientCommand Sequence Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/AddPatientSequenceDiagram.png)
```
Good sequence diagram, helps to make the flow of the program more understandable.

Dri-water commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,209 @@
+The following sequence diagram illustrates how `Parser` parse command for `AddTaskCommand`:
+
+![Parse_Add_Task_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/ParseAddTaskClassDiagram.png)
+
```
I like how sequence diagrams are separated by levels of details and abstraction. Really helps in making diagrams that are easy to understand!

ThisisXXZ commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,209 @@
+
+The following sequence diagram illustrates how `Parser` parse command for `AddTaskCommand`:
+
+![Parse_Add_Task_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/ParseAddTaskClassDiagram.png)
```
Seems that you have mistaken the file name here, the sequence diagram for ``ParseAddTask`` is not properly displayed on your UG website

Dri-water commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,209 @@
+3. **Execution**: The `FindPatientCommand` or `FindTaskCommand` searches for the task.
+
+#### Sequence Diagram
+
```
Very nice how most of the important commands have a sequence diagram to illustrate. Very comprehensive.

Dri-water commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,209 @@
+
+### **State Switching Feature**
+
+The state switching feature allows the system to change its mode of operation between `MAIN_STATE` (for patient management) and `TASK_STATE` (for task management). This allows the software to handle commands differently based on the current state.
```
Usage of state management is explained very clearly, making it easy to understand!

ThisisXXZ commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,209 @@
+
+A closer look on how `AddTaskCommand` is executed is shown below:
+
+![Add_Task_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/AddTaskClassDiagram.png)
```
Same problem here. You may want to correct the file name here.

Dri-water commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,209 @@
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ... | I want to ... | So that I can ... |
```
Would be good if details about what type of nurses could benefit specifically, but overall very comprehensive and clear with the "so that i can..." section!

Dri-water commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,209 @@
 
-* *glossary item* - Definition
+- **Mainstream OS**: Windows, Linux, Unix, MacOS
+- **Personal data**: Data that can be used to identify a person, such as a name and medical records.
```
Good way to ensure everyone is on the same page in what "personal data" refers to. Good job!

Dri-water reviewed (commented):
Overall good job! the DG is very comprehensive, and contains a lot of sequence diagrams! Clearly a lot of effort has been put in! Great work!

ThisisXXZ commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,209 @@
+### Command
+The `commands` package includes the command pattern used in the application to handle user operations. An abstract `Command` class serves as the base for all specific command implementations. Each command class extends `Command` and overrides the `execute()` method to perform its intended action.
+
+![Command_Class_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/CommandClass.png)
```
You may want to change the font size of your diagram for ``CommandClass``. It is a bit difficult to identify the names of each class.

ThisisXXZ commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,209 @@
+
+### Storage
+
+![Storage_Class_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/StorageClassDiagram.png)
```
You may want to modify the arrowhead on a dashed line (from ``StorageFile``, to ``StorageBackup``) to an open (unfilled) arrowhead.

Holy-An commented on docs/diagrams/ParserClass.puml:
```diff
@@ -0,0 +1,60 @@
+
+class UnmarkParser implements CommandParser {
+}
+
+class Parser {
+    +parseCommand(command: String): void
+}
+
+Parser --> AddDeadlineParser
+Parser --> AddParser
+Parser --> AddRepeatParser
+Parser --> AddTodoParser
+Parser --> BackParser
+Parser --> DeleteParser
+Parser --> ExitParser
+Parser --> FindParser
+Parser --> ListParser
+Parser --> MarkParser
+Parser --> SelectParser
+Parser --> UnmarkParser
+
```
Perhaps you could split this up?

naveen42x commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,253 @@
+
+### Parser
+The Parser interface uses a series of classes to implement the various commands.
+![Parser_Class_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/ParserClass.png)
```
Is this non-standard format correct? Should it be using the recommended standard instead? The C in circle is not needed
<img width="907" alt="Screenshot 2024-10-30 at 12 47 36 PM" src="https://github.com/user-attachments/assets/0c6587a6-08fa-4d64-a3a4-0e65c33bab55">

ThisisXXZ commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,209 @@
+
+The following sequence diagram illustrates how the state-switching mechanism works:
+
+![State-Switching Sequence Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/StateSwitchingSequenceDiagram.png)
```
I notice that some of your sequence diagrams use activation bars, but some of them not. You may consider to use a consistent way to structure you sequence diagrams.

ThisisXXZ reviewed (commented):
Overall great DG!

naveen42x commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,253 @@
+### Command
+The `commands` package includes the command pattern used in the application to handle user operations. An abstract `Command` class serves as the base for all specific command implementations. Each command class extends `Command` and overrides the `execute()` method to perform its intended action.
+
+![Command_Class_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/CommandClass.png)
```
Command Class Diagram is difficult to see and would be problematic when converted to pdf. 
Could simplify it.
<img width="902" alt="Screenshot 2024-10-30 at 12 53 14 PM" src="https://github.com/user-attachments/assets/c1ba45b2-dd52-4a4c-9e67-e76db53848fc">

Holy-An commented on docs/diagrams/FindSequenceDiagram.puml:
```diff
@@ -0,0 +1,37 @@
+    FindTaskCommand -> FindTaskCommand : execute()
+    FindTaskCommand -> TaskList : findTasks(keyword)
+    alt No matching tasks found
+        FindTaskCommand -> CommandResult : new CommandResult(MESSAGE_NO_MATCH)
+    else Tasks found
+        FindTaskCommand -> CommandResult : new CommandResult(MESSAGE_SUCCESS)
```
I believe the return arrows should be dotted

Holy-An commented on docs/diagrams/FindSequenceDiagram.puml:
```diff
@@ -0,0 +1,37 @@
+
+alt State is MAIN_STATE
+    FindParser -> Find : extract(line)
+    Find -> FindParser : keyword
```
Same for here, the return arrow should be dotted.

Holy-An commented on docs/diagrams/FindSequenceDiagram.puml:
```diff
@@ -0,0 +1,37 @@
+    end
+else State is TASK_STATE
+    FindParser -> Find : extract(line)
+    Find -> FindParser : keyword
```
Same for here

naveen42x commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,253 @@
+
+The following sequence diagram illustrates how the `AddPatientCommand` is executed:
+
+![AddPatientCommand Sequence Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/AddPatientSequenceDiagram.png)
```
Missing colons which are required to denote the names of an object in sequence diagrams
<img width="822" alt="Screenshot 2024-10-30 at 12 49 52 PM" src="https://github.com/user-attachments/assets/4ed6d5b7-45f1-42e3-aa30-4df9eb14aa84">

naveen42x commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,253 @@
+
+The following sequence diagram illustrates how the `FindPatientCommand` is executed:
+
+![Find_Sequence_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/FindSequenceDiagram.png)
```
Missing colons which are required to denote the names of an object in sequence diagrams

naveen42x commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,253 @@
+
+The following sequence diagram illustrates how the state-switching mechanism works:
+
+![State-Switching Sequence Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/StateSwitchingSequenceDiagram.png)
```
MAIN_STATE & TASK_STATE are incorrectly illustrated and there's a mismatch with the description of how it works
<img width="809" alt="Screenshot 2024-10-30 at 12 54 24 PM" src="https://github.com/user-attachments/assets/ad22de33-da5a-489e-aeb9-8da6aa749cdd">

Holy-An reviewed (commented):
Overall, good work!

coraleaf0602 commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,351 @@
+
+### Architecture
+
+![Architecture](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/ArchitectureDiagram.png)
```
Great job in following the AB3 guide exactly for this section!

coraleaf0602 commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,351 @@
+
+### Parser
+The Parser interface uses a series of classes to implement the various commands.
+![Parser_Class_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/ParserClass.png)
```
I find this slightly difficult to read, could considering using a different format

coraleaf0602 commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,351 @@
+### task
+The `task` package manages all task-related functionality. A `Task` class serves as the base for other task types: `Todo`, `Deadline`, and `Repeat`. Each task type extends `Task` and introduces additional attributes relevant to its behavior. The `TaskList` class maintains a collection of tasks and provides methods to add, delete, find and track the completion rate of tasks.
+
+![Task_Class_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/TaskClassDiagram.png)
```
The task class diagram format could be more appropriate for the class diagrams above

coraleaf0602 commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,351 @@
+
+The following diagram illustrates the structure of the `Hospital` class and its relationships:
+
+![Hospital Class Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/HospitalClassDiagram.png)
```
This is not a sequence diagram - as per the heading

coraleaf0602 commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,351 @@
+
+The following sequence diagram illustrates the **Completion Rate Feature**:
+
+![Completion Rate Sequence Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/CompletionRateSequenceDiagram.png)
```
If the two headings "Show task list for Patient" and "show patient list with complete rate" are two separate scenarios, then could consider splitting the sequence diagram into two for readability

coraleaf0602 commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,351 @@
+
+The following sequence diagram illustrates how the **Tag Patient Feature** operates when adding or updating a tag on a patient:
+
+![TagFeature Sequence Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/TagPatientClassDiagram.png)
```
This is not a sequence diagram - incorrectly labelled

coraleaf0602 commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,351 @@
+
+The following sequence diagram illustrates how the state-switching mechanism works:
+
+![State-Switching Sequence Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/StateSwitchingSequenceDiagram.png)
```
Colon beside objects is needed

AjayShanker-geek committed:
Add interface to Storage

AjayShanker-geek committed:
Update to support Storage interface

AjayShanker-geek committed:
Separate completion rate UI

AjayShanker-geek committed:
Improve user experience that shows all task completion on main screen

AjayShanker-geek committed:
Add JavaDoc to Storage

AjayShanker-geek committed:
Add JavaDoc to Storage

NCF3535 committed:
Improve User Guide

Changes made:
* Added Table of Contents to improve navigation.
* Documented patient management commands: Add, Delete, Find,
  List, and Select, with examples for each command.
* Enhanced explanations for command formats and usage notes.

This commit enhances the User Guide with structured navigation
and detailed documentation on patient management features.

NCF3535 committed:
Merge pull request #104 from NCF3535/docs/chee-fong-user-guide

Improve User Guide

NCF3535 committed:
Merge pull request #103 from NCF3535/docs/chee-fong-ppp

Create Project Portfolio Page

kennethszj committed:
implement tagging for tasks and update ug

kennethszj committed:
Update ppp

lowjunchen committed:
Merge branch 'AY2425S1-CS2113-T11-1:master' into master

kennethszj committed:
Merge pull request #105 from kennethszj/update-ug

implement tagging for tasks, Update User Guide and Add Project Portfolio Page

lowjunchen committed:
Include PPP for my role

Commit:
Update PPP

lowjunchen committed:
Merge pull request #106 from lowjunchen/PPP

Update ppp

Metanyu committed:
Merge pull request #1 from Metanyu/docs/PPP-update

Update PPP

Commit:
minor update in DG - update task classes description and user stories

Metanyu committed:
Merge pull request #107 from Metanyu/docs/PPP-update

Update PPP

AjayShanker-geek committed:
Add Web Config file

AjayShanker-geek committed:
Update Config file for website

AjayShanker-geek committed:
Add Workflow of website

AjayShanker-geek committed:
Fix Config file

AjayShanker-geek committed:
Minor cleanup for website build

AjayShanker-geek committed:
Add Page Level to the website

AjayShanker-geek committed:
Update website ordering

AjayShanker-geek committed:
Fix nav order for the website

AjayShanker-geek committed:
Revert Changes

AjayShanker-geek committed:
Update Config file

AjayShanker-geek committed:
Revert Changes

AjayShanker-geek committed:
Update nav bar

AjayShanker-geek committed:
Update Nav Bar

AjayShanker-geek committed:
Update Nav Bar

AjayShanker-geek committed:
Enable search

AjayShanker-geek committed:
Update page layout and TOC

AjayShanker-geek committed:
Add Dark Mode

AjayShanker-geek committed:
Add my PPP

AjayShanker-geek committed:
Add parent to docs team

AjayShanker-geek committed:
Add javadoc to StageManager

AjayShanker-geek committed:
Update PPP

AjayShanker-geek committed:
Minor Cleanup

AjayShanker-geek committed:
Merge pull request #108 from Metanyu/DG-Update2

minor update in DG and UG

AjayShanker-geek committed:
Remove website dark theme

AjayShanker-geek committed:
Merge pull request #109 from AjayShanker-geek/ajay/JavaDoc-UG-Update

Add JavaDoc to Storage and StateManager & PPP

NCF3535 committed:
Add Javadoc for patient commands

Changes made:
* Added Javadoc for `AddPatientCommand` class and methods
* Added Javadoc for `DeletePatientCommand` class and methods
* Added Javadoc for `ListPatientCommand` class and methods
* Added Javadoc for `FindPatientCommand` class and methods
* Added Javadoc for `SelectPatientCommand` class and methods

These changes enhance documentation consistency and provide clear
descriptions of each patient command's purpose and usage.

NCF3535 committed:
Add Javadoc for date, time, and date-time enums

Changes made:
* Added Javadoc for `DateFormats` enum
* Added Javadoc for `DateTimeFormats` enum
* Added Javadoc for `TimeFormats` enum

These updates improve documentation clarity, providing explanations for
each format and facilitating understanding of the enums' usage.

NCF3535 committed:
Refactor PatientTest.java into patient folder

Changes made:
* Moved `PatientTest.java` to `seedu.duke.patient`
* Updated package declaration accordingly

This refactor organizes test files within their respective folders,
ensuring a consistent project structure.

AjayShanker-geek committed:
Update README.md with project's README.md

AjayShanker-geek committed:
Cleanup UserGuide

AjayShanker-geek committed:
Minor update on portfolio

AjayShanker-geek committed:
Update UG

- Update Storage Class Diagram
- Fix minor issue table view in markdown

AjayShanker-geek committed:
Minor Fix on callout

AjayShanker-geek committed:
Add callout function config file

AjayShanker-geek committed:
Fix Config file

AjayShanker-geek committed:
Update Config

AjayShanker-geek committed:
Update Callouts

AjayShanker-geek committed:
Minor Cleanup

AjayShanker-geek committed:
Minor cleanup on UG

AjayShanker-geek committed:
Merge pull request #111 from AjayShanker-geek/ajay/docs-update

Update DG & UG

AjayShanker-geek committed:
Merge pull request #110 from NCF3535/chee-fong-clean-up1

Enhance Documentation and Structure in Patient and Utility Classes

Metanyu committed:
Merge branch 'AY2425S1-CS2113-T11-1:master' into master

lowjunchen committed:
Change class diagram to integrate color scheme of the group

Commit:
Update user guide for task-related features, minor formatting change: split certain features into main state and task state.

lowjunchen committed:
Change FIXME to TODO to pass Gradle checks

NCF3535 committed:
Update Sequence Diagrams

Changes made:
* Updated sequence diagrams to follow Adressbook3 standards

These changes allow for increased clarity on the sequence
 diagrams.

NCF3535 committed:
Update Sequence Diagrams

Changes made:
* Updated style.puml for standardisation of sequence diagram
colors

These changes allow for increased clarity for developer guide
diagrams.

NCF3535 committed:
Update Class Diagrams

Changes made:
* Updated class diagrams to follow Adressbook3 standards

These changes allow for increased clarity on the class
diagrams.

Metanyu committed:
Merge pull request #117 from NCF3535/docs/chee-fong-DG-update2

Developer Guide Updates

Metanyu committed:
Merge pull request #116 from Metanyu/docs/UG-Update

Update task-related features for UG and minor formatting change

Commit:
Update Diagram

* What changed:
- Updated sequence diagram for add task feature with standardised coloring. Added activation bars for all classes for consistency.
- Updated class diagram for Task and Command classes with standardised coloring. Fixed formatting for less clustered view.

lowjunchen committed:
Merge branch 'master' into colorscheme

lowjunchen committed:
Save images of the edited class and sequence diagrams

lowjunchen committed:
Merge pull request #114 from lowjunchen/colorscheme

Change class diagram to integrate color scheme of the group

NCF3535 committed:
Add and update diagrams for developer guide

Changes made:
* Added updated `.puml` files for HospitalClassDiagram and
  StateManagerClassDiagram.
* Updated corresponding `.png` files in docs/images directory for
  Hospital, StateManager, AddPatient, and DeletePatient diagrams.

These changes improve the documentation by ensuring the diagrams are
accurate and reflect the current system structure.

NCF3535 committed:
Fix exception handling for invalid inputs in DateFormat

Changes made:
* Updated methods in DateFormat to properly throw exceptions for
  invalid inputs using `showToUserException`.

These changes ensure that all invalid inputs are correctly handled,
providing appropriate feedback to the user.

NCF3535 committed:
Update clarity of UserGuide.md

Changes made:
* Updated notes to list supported date and time formats for deadlines.
* Updated notes to inform users of supported relative terms (e.g., today).
* Refined command summary layout and examples for clearer alignment
  with AddressBook-Level3 format.

These changes improve usability by matching AddressBook-Level3 style,
making command references in the User Guide more intuitive.

AjayShanker-geek committed:
Merge pull request #122 from NCF3535/docs/chee-fong-UG-update

Update clarity of UserGuide.md

AjayShanker-geek committed:
Merge pull request #121 from NCF3535/chee-fong-date-time-fix

Fix exception handling for invalid inputs in DateFormat

AjayShanker-geek committed:
Merge pull request #120 from NCF3535/docs/chee-fong-DG-update3

Add and update diagrams for developer guide

AjayShanker-geek committed:
Resize image for better viewing

AjayShanker-geek committed:
Update Architecture Diagram & Storage Class Diagram to follow group format

AjayShanker-geek committed:
Clean up DG

NCF3535 committed:
Refactor class diagrams for improved readability

Changes made:
* Removed exceptions from the state diagrams.
* Updated enumerations to match standard UML notation for consistency.
* Standardized diagram elements for easier interpretation across all
  components.

These updates enhance readability and compliance with UML standards,
making the diagrams more intuitive and aligned with common notation.

kennethszj committed:
update diagrams based on teacher feedback

kennethszj committed:
change diagrams to follow team

kennethszj committed:
change diagram to have dotted line for return

Commit:
git checkout -b Metanyu-docs/Diagram-Update master merge branch 'Metanyu-docs/Diagram-Update'
	Update sequence diagram according to TA's feedback add colons.
	Additional activation bars for all classes for consistency.
	Changed coloring to meet standard.

	Update class diagram for Task and Command classes with
	standardised coloring. Fixed formatting for less clustered view.

Commit:
Resolve conflicts by taking newer version from fork

Metanyu committed:
Merge pull request #3 from Metanyu/master

Resolve conflicts

Metanyu committed:
Merge pull request #118 from Metanyu/docs/Diagram-Update

Update Diagram

AjayShanker-geek committed:
Merge pull request #123 from AjayShanker-geek/ajay/fix-diagrams

Update diagrams

kennethszj committed:
Merge pull request #124 from kennethszj/update-diagrams

update diagrams based on teacher feedback

NCF3535 committed:
Merge pull request #125 from NCF3535/docs/chee-fong-DG-update4

Refactor class diagrams for improved readability

kennethszj committed:
Add arrow to show inheritance in class diagram

lowjunchen committed:
Resolve Issue #126

AjayShanker-geek committed:
Disable NAV in website

AjayShanker-geek committed:
Add custom.scss

AjayShanker-geek committed:
Disable Nav

AjayShanker-geek committed:
Fix UserGuide TOC

lowjunchen committed:
Merge pull request #128 from lowjunchen/bugfix

Resolve Issue #126

lowjunchen committed:
Update images based on scaling

lowjunchen committed:
Merge pull request #129 from lowjunchen/bugfix

Update images based on scaling

Commit:
What changed:
- Reformat User Guide for better view on the website/
- Added user guide contribution for PPP.

lowjunchen committed:
Revert "Update images based on scaling"

Commit:
What changed:
- Update task description to be case insensitive. Now, finding task or duplicated task will be case-insensitive.
- Update test coverage to be more throughout for TaskList and Task-related Commands.

Metanyu committed:
Merge pull request #130 from Metanyu/docs/PPP-Update2

PPP Update & User Guide minor reformart

Commit:
Deleted unused imports in TaskCommandTest.

kennethszj committed:
modify diagram to show members

NCF3535 committed:
Merge pull request #131 from AY2425S1-CS2113-T11-1/revert-129-bugfix

Revert "Update images based on scaling"

kennethszj committed:
Merge pull request #127 from kennethszj/update-diagram

correct mistakes in class diagrams

NCF3535 committed:
Merge pull request #132 from Metanyu/test/UpdateTestCoverage

Update test coverage for tasks

NCF3535 committed:
Refactor diagrams for improved readability

Changes made:
* Resized diagrams to ensure it aligns with the font sizes in the Developer
Guide

These updates enhance readability,making the diagrams more intuitive
for users to read.

NCF3535 committed:
Fix return arrow for delete command

Changes made:
* Changed standard arrow to dotted arrows for all return arrows

These updates enhance readability,making the diagrams more intuitive
for users to read.

NCF3535 committed:
Fix return arrow for delete command

Changes made:
* Changed standard arrow to dotted arrows for all return arrows

These updates enhance readability,making the diagrams more intuitive
for users to read.

NCF3535 committed:
Merge pull request #133 from NCF3535/docs/chee-fong-DG-update5

Refactor diagrams for improved readability

lowjunchen committed:
Update the font size, scale and save the images

lowjunchen committed:
Merge pull request #134 from lowjunchen/updatediagrams

Update the font size, scale and save the images

lowjunchen committed:
update parser class font size as it is still too small

kennethszj committed:
Merge pull request #135 from lowjunchen/updatediagrams

update parser class font size as it is still too small

NCF3535 committed:
Refactor diagrams for improved readability

Changes made:
* Resized diagrams to ensure it aligns with the font sizes in the Developer
Guide

These updates enhance readability,making the diagrams more intuitive
for users to read.

NCF3535 committed:
Merge pull request #136 from NCF3535/docs/chee-fong-DG-update6

Refactor diagrams for improved readability

AjayShanker-geek committed:
Merge branch 'master' of github.com:AjayShanker-geek/tp

AjayShanker-geek committed:
Update Storage diagram font

NCF3535 committed:
Refactor diagrams for improved readability

Changes made:
* Resized diagrams to ensure it aligns with the font sizes in the Developer
Guide

These updates enhance readability,making the diagrams more intuitive
for users to read.

NCF3535 committed:
Merge pull request #137 from NCF3535/master

Refactor diagrams for improved readability

NCF3535 committed:
Fix date and time parsing error handling

Previously, date and time parsing had an error where it rejected valid inputs.

Changes made:
* Ensure exception is shown correctly
* Improved error handling to only show error messages after all format checks fail.

These changes improve the user experience by providing clear feedback when inputs are invalid.

NCF3535 committed:
Handle invalid index inputs for delete command in Parser

Previously, the delete command did not handle non-integer inputs
correctly, which could lead to errors and confusion for users.

Changes made:
* Added `IllegalValueException` handling specifically to the `delete`
  case in `Parser` to manage invalid index inputs.
* Updated the `delete` command’s error handling to display specific
  messages for invalid indices and log warnings.

These changes improve input validation for the `delete` command,
ensuring clearer error messages and preventing unexpected behavior
when a non-integer index is entered.

kennethszj committed:
fix bug where adding a task silently fails

kennethszj committed:
Merge pull request #205 from NCF3535/chee-fong-bug-fix1

Fix date and time parsing error handling

kennethszj committed:
Merge pull request #206 from NCF3535/chee-fong-bug-fix2

Handle invalid index inputs for delete command in Parser

kennethszj committed:
add illegal value exception to select command

NCF3535 committed:
Merge pull request #208 from kennethszj/fix-invalid-index

add illegal value exception to select command

NCF3535 committed:
Merge pull request #207 from kennethszj/fix-add-command-in-task-state

fix bug where adding a task silently fails

kennethszj committed:
change completion rate to show no tasks available instead of 0 or 100 percent

kennethszj committed:
fix checkstyle

AjayShanker-geek committed:
Merge branch 'AY2425S1-CS2113-T11-1:master' into master

AjayShanker-geek committed:
Revert change by removing custom css

lowjunchen committed:
Fix the following issues
- Mark/unmark/delete command crashes when the index is a string
- When adding a patient, typing /tag which is the tag flag, and not adding a description after the flag without a space makes the flag grouped as the patient's name.
- When adding a repeat task, typing /every which is the flag, and not adding a description after the flag without a space makes the flag grouped as the task's name.
- Solves issue 184, 182, 152, 148, 146, 145, 143,

NCF3535 committed:
Add log to file system

Previously, all logs were shown on console, with logs being below `SEVERE`
not being shown

Changes made:
*Created `LoggerConfig` to establish a new logging system
*Set up `FileHandler` to log all levels to `meditask.log` in `logs` directory
*Configured `ConsoleHandler` to show only SEVERE messages in console
*Removed default handlers to ensure only configured logging is applied

This update introduces a structured logging system from scratch, allowing all
log messages to be saved in a dedicated log file while limiting console output to
severe errors only. This enhances tracking for detailed logs and keeps the
console focused on critical issues.

NCF3535 committed:
Update severity level to warning

Previously, severity levels were all set to severe, which were too high for
simple issues

Changes made:
*updated severity level from `SEVERE` TO `WARNING`

This change adjusts the logging level to WARNING, allowing less critical
issues to be logged and improving insight into application activity without
overloading the console or log files with excessive severity.

kennethszj committed:
Merge pull request #211 from NCF3535/chee-fong-logger

Add file-based logging and adjust severity level to WARNING

AjayShanker-geek committed:
Improve help message for both patient state and task state

AjayShanker-geek committed:
Update help images

AjayShanker-geek committed:
Clean up file

AjayShanker-geek committed:
Fix Flags from Check-style

AjayShanker-geek committed:
Remove exceptions in Storage Class Diagram

Metanyu committed:
Merge pull request #210 from lowjunchen/pe-d

Fix the following issues

Metanyu committed:
Merge pull request #212 from AjayShanker-geek/ajay/PE-add-examples-help

Add Examples to Help Command

Metanyu committed:
Merge pull request #209 from kennethszj/resolve-pe-issues

fix cosmetic bug

Commit:
Fix #189

What changed: Fix formatting relationship between TaskList and Task. Since there can be any amount of Task objects, this can just be `*` (this was previous mislabed as `0...*`.

Commit:
Change scaling for better view

Metanyu committed:
Merge pull request #213 from AjayShanker-geek/ajay/PE-fix-storage-diagram

Update Storage Diagram

Commit:
Fix #198
What changed:

- Marking a task/unmarking a task that has already been marked/unmarked will now give a warning to user.

kennethszj committed:
Merge pull request #215 from Metanyu/issue/189.tasklist-classdiagram-update

TaskList Class Diagram Update

kennethszj committed:
Merge pull request #216 from Metanyu/issue/198.duplicate-marking-task-fix

Add repeated marking/unmarking task warning

NCF3535 committed:
Fix Add Patient Sequence Diagram

Changes made:
* Removed unecessary wording to increase readability
* Fixed issue where back arrow skips Parser and AddParser
before returning
* Ensure that object creation is accurately shown

These updates enhance readability,making the diagrams more intuitive
for users to read.

NCF3535 committed:
Fix Delete Patient Sequence Diagram

Changes made:
* Removed unecessary wording to increase readability
* Fixed issue where alt overlaps with activation bar
* Remove return void
* Ensure that object creation is accurately shown

These updates enhance readability,making the diagrams more intuitive
for users to read.

NCF3535 committed:
Fix Hospital Class Diagram Multiplicity

Changes made:
* Updated Multiplicity to standard notation

These updates enhance readability,making the diagrams more intuitive
for users to read.

kennethszj committed:
Merge pull request #217 from NCF3535/docs/DG-update7

Fix Developer Guide Sequence and Class Diagrams

kennethszj committed:
change UG to reflect the correct number of tags a patient can have

NCF3535 committed:
Fix notes link in User Guide

Changes made:
* Fix Capitalisation error which result in broken links

This update should help improve calrity for users.

kennethszj committed:
Add find tasks/patients based on tag

NCF3535 committed:
Add sample outputs for User Guide

Changes made:
* Added sample outputs to Patient commands in User Guide

This update should help improve clarity for users, allowing them to verify the
outputs on their screen.

kennethszj committed:
modify test case to match new find command output

Commit:
Fix #193
Fix #196
Fix #199
Fix #203

What changed:
- Deleted ParseAddTaskSequenceDiagram altogether as it seems redundant to describe the `AddTaskCommand` implementation.
- Remove extra wording to make AddTaskSequenceDiagram smaller in size and easier to see.
- Correctly places AddTaskCommand and Task box to where its creation is.
- Fix the return value to be returned to parser before return to MediTask
- Fixed DG to accommdate deletion of ParseAddTaskSequenceDiagram and minor edit in AddTaskCommand impementation explanation.

kennethszj committed:
modify failing testcases

kennethszj committed:
update UserGuide

Commit:
Fix #179
What changed:
- Deadline and Repeat now handle missing arguments on creation.
- Separate MissingTaskArgument into its own exception class.
- Changed default value for duration result to be in line with Repeat catch.
- Added tests for task creation factory to check handle of many task creation error.

Commit:
Edit test to work with updated error missing task handling

NCF3535 committed:
Merge pull request #218 from NCF3535/docs/chee-fong-UG-update2

Updated User Guide

kennethszj committed:
Merge pull request #219 from kennethszj/resolve-pe-d-issues

Resolve issue #149 and #168

Metanyu committed:
Merge pull request #220 from Metanyu/issue/sequence-diagram-update

AddTaskCommand Sequence Diagram + DG update

Metanyu committed:
Merge pull request #221 from Metanyu/issue/179.repeat-nothing-fix

Repeat and Deadline update to handle missing arguments

NCF3535 committed:
Fix inaccurate activation bar

Previously, there were broken activation bars on items that were still
on the same instance.

Changes made:
* Join back activation bars that are the same instance
* Ensure activation bars are not unncessarily long

These changes will help in readibility and calrity for the user.

NCF3535 committed:
Fix broken activation bar

Previously, there were broken activation bars on items that were still
on the same instance.

Changes made:
* Join back activation bars that are the same instance

These changes will help in readibility and clarity for the user.

NCF3535 committed:
Fix missing return arrow

Changes made:
* Add back "void" arrow

These changes will help in readibility and calrity for the user.

NCF3535 committed:
Merge pull request #222 from NCF3535/docs/chee-fong-DG-update7

Fix inaccurate activation bar

NCF3535 committed:
Update Project Portfolio Page

Changes made:
* Documented latest contributions to the Developer Guide
and User Guide
*Split the commands out to explain each one of them in greater
detail

This commit adds a comprehensive Project Portfolio Page,
highlighting key contributions and community involvement.

NCF3535 committed:
Fix spelling in PPP

Changes made:
* Fixed spelling from "enum" to Enumerations

This will help improve readability and clarity for users.

NCF3535 committed:
Merge pull request #223 from NCF3535/docs/chee-fong-ppp2

Update Project Portfolio Page

AjayShanker-geek committed:
Add deserializer class to remove duplicates hospital class is loaded

AjayShanker-geek committed:
Update Test Case

AjayShanker-geek committed:
Enable StorageDeserializer as jackson module

AjayShanker-geek committed:
Update Storage Diagram

AjayShanker-geek committed:
Updated DG:

- Add feature that checks for duplication
- Add more details about logger

AjayShanker-geek committed:
Fix error from check-style

lowjunchen committed:
TagPatientClassDiagram: The label is now closer to the end of the arrow #202
FindPatientCommandDiagram / ParserClass: Increased the size #201, #176

lowjunchen committed:
Include my photo in about us page

AjayShanker-geek committed:
Merge pull request #224 from AjayShanker-geek/ajay/PE-fix-duplicate

Fix Duplications on JSON

NCF3535 committed:
Align AboutUs.md formatting

Changes made:
*If a picture has not been inserted, the github avatar is used as image placeholder
*Aligned everyone's portfolio to the PPP

This ensures standardisation and easy contact if others are interested.

NCF3535 committed:
Fix broken links

Changes made:
* fixed issue where links ported over from README.md from
docs did not work on the main repo page

This will ensure that users can easiliy navigate to our Developer Guide,
User Guide and About Us pages.

lowjunchen committed:
Address NCF3535 comments

NCF3535 committed:
Fix broken links

Changes made:
* fixed issue where links did not work

This will ensure that users can easiliy navigate our About Us pages.

NCF3535 committed:
Merge pull request #227 from NCF3535/docs/chee-fong-readme1

Fix broken links

lowjunchen committed:
removed redundant box

lowjunchen committed:
remove scale which caused diagram to be cut off

lowjunchen committed:
Update photo

kennethszj committed:
Merge pull request #226 from NCF3535/docs/chee-fong-about-me

Align AboutUs.md formatting

lowjunchen committed:
Implement missing activator bars and relevant details such as return arrows and colon for instantiation

lowjunchen committed:
Update findsequencediagram

lowjunchen committed:
Prevent overlapping of text for alt block

AjayShanker-geek committed:
Add support to recalculate completion rate when loading data

AjayShanker-geek committed:
Add JavaDoc

lowjunchen committed:
Remove font size

lowjunchen committed:
Merge branch 'master' into pe-d-diagrams

lowjunchen committed:
Merge pull request #225 from lowjunchen/pe-d-diagrams

Fix issues

AjayShanker-geek committed:
Merge pull request #228 from AjayShanker-geek/ajay/PE-fix-completion-cal

Recalculation Completion Rate

lowjunchen committed:
edit portfolio page url

NCF3535 committed:
Fix PPP formating

Changes made:
* Edited link to reference a more accuate PR
* Ensure bold is consistent throughout the PPP
* Remove unnecessary spaces

This will help improve clarity and consistency of the PPP.

NCF3535 committed:
Remove unnecessary spaces

Changes made:
* Remove unnecessary spaces

This will help improve clarity and consistency of the PPP.

kennethszj committed:
Merge pull request #230 from lowjunchen/abousus-edit

edit portfolio page url

NCF3535 committed:
Merge pull request #231 from NCF3535/docs/chee-fong-ppp3

Update clarity in PPP

NCF3535 committed:
Fix Bolding issues

Changes made:
* Ensure bold is consistent throughout the PPP

This will help improve clarity and consistency of the PP

AjayShanker-geek committed:
Add removed exception message

NCF3535 committed:
Merge pull request #232 from NCF3535/docs/chee-fong-ppp4

Fix Bolding issues

AjayShanker-geek committed:
Merge pull request #235 from AjayShanker-geek/master

Add removed exception message

NCF3535 committed:
Add example test cases

Changes made:
* Added example test cases to the developer guide to
improve clarity
* Added examples of test cases that can cause an
error

This will help improve clarity on what to expect with some of the
commands in the Developer Guide

NCF3535 committed:
Merge pull request #236 from NCF3535/docs/chee-fong-DG-update8

Add example test cases

NCF3535 committed:
Fix Bolding issues

Changes made:
* Ensure bold is consistent throughout the PPP

This will help improve clarity and consistency of the PPP

lowjunchen committed:
update syntax

NCF3535 committed:
Merge pull request #237 from NCF3535/ppp-format-fix

Fix Bolding issues

lowjunchen committed:
Merge branch 'master' into PPP-edit-syntax

lowjunchen committed:
Merge pull request #238 from lowjunchen/PPP-edit-syntax

update syntax

AjayShanker-geek committed:
Upscale Storage Diagram

AjayShanker-geek committed:
Merge pull request #239 from AjayShanker-geek/ajay/PE-upscale-diagram

Upscale Storage Diagram

kennethszj committed:
Fix for Empty or Whitespace-only Input Causing Program Crash

kennethszj committed:
Merge pull request #241 from kennethszj/bug-fix

Fix for Empty or Whitespace-only Input Causing Program Crash

Commit:
What changed:
- Correct the name of .jar file
- Fix formatting for `adding a deadline task` feature's notes.

Commit:
Fix #242
What changed:
- Added a quick fix to check for blank patient name before adding

Commit:
Fix #243
Tag now checks for blank value before executing.

Commit:
Update PPP

NCF3535 committed:
Merge pull request #244 from Metanyu/docs/UG-Update2

Minor UG update

NCF3535 committed:
Merge pull request #245 from Metanyu/issue/242.blank-patient-name

Fix blank patient name

NCF3535 committed:
Merge pull request #246 from Metanyu/issue/243.blank-tag

Fix blank tag

NCF3535 committed:
Merge pull request #247 from Metanyu/docs/PPP-Update3

Update PPP

NCF3535 committed:
Fix incorrect PR reference

Changes made:
*Ensured referenced PR is correct.

This helps with user clarity when reading the PPP.

NCF3535 committed:
Merge pull request #248 from NCF3535/docs/chee-fong-ppp5

Fix incorrect PR reference

AjayShanker-geek committed:
Update PPP

AjayShanker-geek committed:
Merge pull request #5 from AjayShanker-geek/ajay/update-PPP

Update PPP

AjayShanker-geek committed:
Improve layout

AjayShanker-geek committed:
Merge pull request #6 from AjayShanker-geek/ajay/update-PPP

Improve layout

AjayShanker-geek committed:
Update Format

AjayShanker-geek committed:
Merge pull request #7 from AjayShanker-geek/ajay/update-PPP

Update Format

AjayShanker-geek committed:
Disable Nav

AjayShanker-geek committed:
Merge pull request #8 from AjayShanker-geek/ajay/update-PPP

Disable Nav

AjayShanker-geek committed:
Enable Nav bar

AjayShanker-geek committed:
Merge pull request #250 from AjayShanker-geek/ajay/update-PPP

Update PPP

AjayShanker-geek committed:
Merge branch 'master' of github.com:AjayShanker-geek/tp

AjayShanker-geek committed:
Rename PPP

AjayShanker-geek committed:
Merge pull request #251 from AjayShanker-geek/ajay/rename-PPP

Rename PPP
